### DISCUSSION TITLE
How to use s3 in other hosting (not aws)

### DISCUSSION THREAD

**Discussion by timakon** [serverpod/serverpod#1100] [Q&A] [answered]
I wanna to use yandex cloud storage (compatible with s3). What i need to do? Remove aws folder? Change some files in aws folder?
Mb u can show an example not with aws but with smth else

**Comment by vlidholt** [comment, marked as answer]
Right now, I think the easiest way to accomplish this is to make a copy of the https://github.com/serverpod/serverpod/tree/main/integrations/serverpod_cloud_storage_s3 package and tweak it so that it works with your provider.

**Reply by abdirahmanabdullahi** [reply]
Hello. Am trying to use locally hosted minIO for object storage. Please can you help with the tweaking. Am kinda lost. Thank you so much. And nice work with serverpod. I love it

**Reply by mathis6787** [reply]
Check this out  **PR:** https://github.com/serverpod/serverpod/pull/4200

**Comment by iampopal** [comment]
We also need this

**Comment by mathis6787** [comment]
Hi everyone 👋

I had the same issue when trying to use Serverpod with MinIO (and other S3-compatible providers).  
Good news — I opened a PR to add proper host override support to `serverpod_cloud_storage_s3` so it can work with MinIO and likely other S3-like providers.

🔗 **PR:** https://github.com/serverpod/serverpod/pull/4200

I hope it will get merged soon 🙏

---

# 🧪 How to use it right now (until merged)

Add this to your `pubspec.yaml` under `dependency_overrides`:

```yaml
dependency_overrides:
  serverpod_cloud_storage_s3:
    git:
      url: https://github.com/mathis6787/serverpod.git
      path: integrations/serverpod_cloud_storage_s3
      ref: feature/add-s3-host-override

Then make sure your Serverpod versions match, for example:
serverpod: 3.0.0-alpha.3
serverpod_test: 3.0.0-alpha.3
serverpod_auth_server: 3.0.0-alpha.3

Example in your server.dart: 

  pod.addCloudStorage(s3.S3CloudStorage(
    serverpod: pod,
    storageId: 'item-images',
    endpointUrl: 'http://localhost:9000',
    host: 'localhost:9000',
    public: false,
    region: 'us-east-1', 
    bucket: 'your-bucket-name, 
    publicHost: 'localhost:9000/your-bucket-name',
    useHttps: false,
  ));


📌 Extra note

If this PR gets accepted, I plan to work on improving S3CloudStorage further.
Right now the implementation is limited — especially private buckets — and it’s missing important features such as 
 
•	Generating presigned URLs (GET / PUT / DELETE) to securely access private objects without exposing buckets.



I’d be happy to continue contributing these improvements if the maintainers are open to it.
